collection-scripts/gather_istio

```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+# Copyright 2025 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source $(dirname "$0")/common.sh
+BASE_COLLECTION_PATH="/must-gather/istio"
+
+# Get Istio, sail operator, kiali, gateway.networking.k8s.io, inference.networking.x-k8s.io and inference.networking.k8s.io group CRDs
+function getCRDs() {
+  local result=()
+  local output
+  output=$(oc get crds -o custom-columns=NAME:metadata.name --no-headers | grep -e '\.istio\.io' -e '\.sailoperator\.io' -e '\.kiali\.io' -e '\.gateway\.networking\.k8s\.io' -e 'inference\.networking\.x-k8s\.io' -e 'inference\.networking\.k8s\.io')
```

Comment:
Why use grep here and not a oc template?

Reply:
The main reason I see is that there are multiple API groups within/ending with `.istio.io` and it's easier to get all CRDs for all of them with `grep` rather than having to list each istio API group explicitly within a template (e.g. `{{- if eq .spec.group "networking.istio.io" "security.istio.io" "telemetry.istio.io" ... -}}`) and potentially miss any new ones in the future. If there's a particular advantage to using an oc template or an easy way to do substring matching though I'm all ears.